tests/update_state_test.py

```diff
@@ -0,0 +1,231 @@
+'''This module runs tests for the three functions in the update stare section of the game loop:
+1) check_guess 2) check_win 3) update_state'''
+import sys
+sys.path.append("..")
+from game import Game
+
+
+
+game = Game()
+
+class TestCheckGuess:
+    'Check_guess is responsible for checking the guessed word and identifying which deck it belongs to.'
+    def test_correct_guess(self):
+        'test if a guess can equal a word in cards resulting in card.flipped set equal to True'
+        # Create a correct guess
+        game.guess = game.board.cards[0].word
+
+        game.check_guess()
+
+
+        assert game.board.cards[0].flipped is True
+
+    def test_guessed_correct_word(self):
+        'test if the guessed word belongs to the team that picked it'
+        # game.guess is set equal to the first word that has the same color as the teams color
+        game.guess = [card.word for card in game.board.cards if card.color == game.team][0]
+
+        game.check_guess()
+
+        assert game.guess_value == 1
+
+    def test_guessed_opponents_word(self):
+        'test if the guessed word belongs to the other team'
+        game.team = "red"
+        # game.guess is the first word thats color is the opposite teams word
+        game.guess = [card.word for card in game.board.cards if card.color == "blue"][0]
+
+        game.check_guess()
+
+        assert game.guess_value == 2
+
+    def test_guessed_neutral_word(self):
+        'test if the guessed word belongs to the neutral stack'
+        # game.guess is the first neutral word
+        game.guess = [card.word for card in game.board.cards if card.color == "yellow"][0]
+
+        game.check_guess()
+
+        assert game.guess_value == 3
+
+    def test_guessed_death_word(self):
+        'test if the guessed word is the death card'
+        # game.guess is the death word
+        game.guess = [card.word for card in game.board.cards if card.color == "black"][0]
+
+        game.check_guess()
+
+        assert game.guess_value == 4
+
+    def test_code_number_decrease(self):
+        'test if code_number gets decreased when the correct answer is guessed'
+        game.code_number = 3
+        game.guess = ""
```

Comment:
This assumes the function works a certain way. You know internally that this function will skip setting `guess_value` if you set `guess` to a word that can't appear. Since you are using internal knowledge about the function, you are constraining your ability to change the function implementation, even in ways that don't change the final outputs. You could avoid having to do this by combining a few of your tests. Maybe have four tests built around the four possible results of a guess. Then check all three state variables that `check_guess()` sets in each of the four tests. The first test could look like this.

```py
def test_correct_guess():
    game = Game()
    game.code_number = 3
    game.guess = [card.word for card in game.board.cards if card.color == game.team][0]

    game.check_guess()

    assert [card.flipped for card in game.board.cards if card.word == game.guess][0] is True
    assert game.guess_value == 1
    assert game.code_number == 2
```

See if this type of change would make sense in your other groups of tests as well.